public/app/features/alerting/unified/api/grafana.ts

```diff
@@ -1,13 +1,90 @@
 import { getBackendSrv } from '@grafana/runtime';
-import { ContactPointsState, NotifierDTO } from 'app/types';
+import { ContactPointsState, NotifierDTO, ReceiversStateDTO } from 'app/types';
 
 export function fetchNotifiers(): Promise<NotifierDTO[]> {
   return getBackendSrv().get(`/api/alert-notifiers`);
 }
 
+const hasArrayIndex = (name: string) => name.indexOf('[') !== -1;
+
+export const isValidIntegrationType = (integrationName: string): boolean =>
+  hasArrayIndex(integrationName) ? /\w(\[((\d*))])$/.test(integrationName) : true;
```

Comment:
I think we could simplify this to a single regular expression like `^(\w+)(\[\d+\])?$` where we just mark the array index part entirely as optional so we don't need the `hasArrayIndex` check and we can extract both the `integrationType` and `index` from the capture groups.

Something along the lines of 

```js
export const parseIntegrationName = (integrationName: string) => {
  const matches = integrationName.match(/^(\w+)(\[\d+\])?$/);
  if (!matches) {
    return;
  }

  return {
    type: matches[1],
    index: matches[2],
  };
};
```

Reply:
makes sense! @gillesdemey 👌